src/Parser.js

```diff
@@ -507,6 +507,20 @@ Parser.prototype.parseImplicitGroup = function() {
                 body: new ParseNode("ordgroup", body, this.mode),
             }, this.mode);
         }
+    } else if (func === "$") {
+        if (this.mode === "math") {
+            throw new ParseError("$ within math mode");
+        }
+        this.consume();
+        const outerMode = this.mode;
+        this.switchMode("math");
+        const body = this.parseExpression(false, "$");
+        this.expect("$", true);
+        this.switchMode(outerMode);
+        return new ParseNode("styling", {
+            style: "text",
+            value: body,
+        }, "math");
```

Comment:
Added a couple of unit tests, including on the MathML generation side, as well as a screenshot test (which also tests spacing).